docs/reference/commandline/service_create.md

```diff
@@ -445,6 +447,36 @@ The following options can only be used for bind mounts (`type=bind`):
       </ul>
     </td>
   </tr>
+  <tr>
+    <td><b>bind-readonly-nonrecursive</b> or <b>bind-ro-nonrecursive</b></td>
```

Comment:
I'm considering if we should (soft) deprecated `bind-nonrecursive`, and instead have a "recursive mounts" option that takes a (non-boolean) value, e.g. something like;

- `recursive-mounts=enabled` (default)
- `recursive-mounts=disabled`
- `recursive-mounts=readonly` (should error if `bind-propagation` is set by user (to anything other than `rprivate`), but perhaps handled by API)
- `recursive-mounts=force-readonly` (should error if `bind-propagation` is set by user (to anything other than `rprivate`), but perhaps handled by API)

Reply:
What about this?
```go
                case "bind-recursive":
                        valS := val
                        // Allow boolean as an alias to "enabled" or "disabled"
                        if b, err := strconv.ParseBool(valS); err == nil {
                                if b {
                                        valS = "enabled"
                                } else {
                                        valS = "disabled"
                                }
                        }
                        switch valS {
                        case "enabled": // read-only mounts are recursively read-only if Engine >= v25 && kernel >= v5.
12
                                // NOP
                        case "disabled": // alias of bind-nonrecursive=true
                                bindOptions().NonRecursive = true
                        case "readonly-classic": // conforms to Docker v24; read-only mounts are recursively mounted bu
t not recursively read-only
                                bindOptions().ReadOnlyNonRecursive = true
                                mount.ReadOnly = true
                        case "readonly-force": // force recursively read-only, or raise an error
                                bindOptions().ReadOnlyForceRecursive = true
                                bindOptions().Propagation = mounttypes.PropagationRPrivate
                                mount.ReadOnly = true
                        default:
                                return fmt.Errorf("invalid value for %s: %s (must be \"enabled\", \"disabled\", \"reado
nly-classic\", or \"readonly-force\")",
                                        key, val)
                        }
```